In [1]:
import pandas as pd

df = pd.read_csv (r'/Users/joshuazalkan/Downloads/shots_data.csv')
print (df)

       team     x     y  fgmade
0    Team A -23.1   3.5       0
1    Team A   0.0  25.1       1
2    Team A   0.5   1.0       1
3    Team A  -5.6   5.2       0
4    Team A   4.0  14.5       1
..      ...   ...   ...     ...
555  Team B   7.8  24.7       0
556  Team B  10.1   1.4       1
557  Team B -17.8  -0.1       0
558  Team B  22.6   2.9       0
559  Team B  -0.7  14.7       1

[560 rows x 4 columns]


In [259]:
def shot_zone(row):
    if abs(row['x']) < 22 and row['y'] <= 7.8:
        return '2PT'
    elif row['x']**2 + row['y']**2 <= 23.75**2 and row['y']>=7.8:
        return '2PT'
    elif abs(row['x']) >22 and row['y'] < 7.8:
        return 'C3'
    else:
        return 'NC3'

In [260]:
df['shotzone'] = df.apply(lambda row: shot_zone(row), axis=1)

In [261]:
df

team     x     y  fgmade shotzone
0    Team A -23.1   3.5       0       C3
1    Team A   0.0  25.1       1      NC3
2    Team A   0.5   1.0       1      2PT
3    Team A  -5.6   5.2       0      2PT
4    Team A   4.0  14.5       1      2PT
..      ...   ...   ...     ...      ...
555  Team B   7.8  24.7       0      NC3
556  Team B  10.1   1.4       1      2PT
557  Team B -17.8  -0.1       0      2PT
558  Team B  22.6   2.9       0       C3
559  Team B  -0.7  14.7       1      2PT

[560 rows x 5 columns]

In [262]:
team_a_fga = len(df[df['team']== 'Team A'])

In [263]:
team_a_fga

280

In [264]:
team_a_2PT = len(df[(df['team'] == 'Team A') & (df['shotzone'] == '2PT')])

In [265]:
team_a_2PT

170

In [266]:
team_a_C3 = len(df[(df['team'] == 'Team A') & (df['shotzone'] == 'C3')])

In [267]:
team_a_C3

20

In [268]:
team_a_NC3 = len(df[(df['team'] == 'Team A') & (df['shotzone'] == 'NC3')])

In [269]:
team_a_NC3

90

In [270]:
team_a_2PT + team_a_C3 + team_a_NC3 == team_a_fga

True

In [271]:
team_a_2PT/team_a_fga * 100

60.71428571428571

In [272]:
team_a_C3/team_a_fga * 100

7.142857142857142

In [273]:
team_a_NC3/team_a_fga * 100

32.142857142857146

In [274]:
60.71+7.14+32.14

99.99

The shot distribution of Team A is as follows:

    2PT - 60.71% 
    C3 - 7.14%
    NC3 - 32.14%

In [275]:
team_a_2PT_made = len(df[(df['team'] == 'Team A') & (df['shotzone'] == '2PT') & (df['fgmade'] == 1)])

In [276]:
team_a_2PT_made

68

In [277]:
team_a_C3_made = len(df[(df['team'] == 'Team A') & (df['shotzone'] == 'C3') & (df['fgmade']==1)])

In [278]:
team_a_C3_made

8

In [279]:
team_a_NC3_made = len(df[(df['team'] == 'Team A') & (df['shotzone'] == 'NC3') & (df['fgmade']==1)])

In [280]:
team_a_NC3_made

29

In [281]:
team_a_3PT_made = 8+29

In [282]:
team_a_3PT_made

37

In [283]:
team_a_fgm = len(df[(df['team']=='Team A') & (df['fgmade']==1)])

In [284]:
team_a_fgm

105

In [285]:
team_a_eFG = ((team_a_fgm + (.5 * team_a_3PT_made))/team_a_fga)

In [286]:
team_a_eFG

0.44107142857142856

Team A's total eFG% is 44.11%

In [287]:
team_a_2PT_efg = ((team_a_2PT_made)/team_a_2pt)

In [288]:
team_a_2PT_efg

0.4

In [289]:
team_a_C3_efg = ((team_a_C3_made + (.5 * team_a_C3_made))/team_a_C3)

In [290]:
team_a_C3_efg

0.6

In [291]:
team_a_NC3_efg = ((team_a_NC3_made + (.5 * team_a_NC3_made))/team_a_NC3)

In [292]:
team_a_NC3_efg

0.48333333333333334

On each respective shot, Team A's eFG% was:

    2PT - 40%
    C3 - 60%
    NC3 - 48.33%

In [293]:
team_b_fga = len(df[df['team']== 'Team B'])

In [294]:
team_b_fga

280

In [295]:
team_b_2PT = len(df[(df['team'] == 'Team B') & (df['shotzone'] == '2PT')])

In [296]:
team_b_2PT

163

In [297]:
team_b_C3 = len(df[(df['team'] == 'Team B') & (df['shotzone'] == 'C3')])

In [298]:
team_b_C3

21

In [299]:
team_b_NC3 = len(df[(df['team'] == 'Team B') & (df['shotzone'] == 'NC3')])

In [300]:
team_b_NC3

96

In [301]:
team_b_2PT + team_b_C3 + team_b_NC3 == team_b_fga

True

In [302]:
team_b_2PT/team_b_fga * 100

58.214285714285715

In [303]:
team_b_C3/team_b_fga * 100

7.5

In [304]:
team_b_NC3/team_b_fga * 100

34.285714285714285

In [305]:
58.21 + 7.5 + 34.29

100.0

The shot distribution of Team B is as follows:

    2PT - 58.21%
    C3 - 7.5%
    NC3 - 34.29%

In [306]:
team_b_2PT_made = len(df[(df['team'] == 'Team B') & (df['shotzone'] == '2PT') & (df['fgmade']==1)])

In [307]:
team_b_2PT_made

75

In [308]:
team_b_C3_made = len(df[(df['team'] == 'Team B') & (df['shotzone'] == 'C3') & (df['fgmade']==1)])

In [309]:
team_b_C3_made

5

In [310]:
team_b_NC3_made = len(df[(df['team']== 'Team B') & (df['shotzone']== 'NC3') & (df['fgmade']==1)])

In [311]:
team_b_NC3_made

35

In [312]:
team_b_fgm = len(df[(df['team']=='Team B') & (df['fgmade']==1)])

In [313]:
team_b_fgm

115

In [314]:
team_b_3PT_made = team_b_C3_made + team_b_NC3_made

In [315]:
team_b_3PT_made

40

In [316]:
team_b_eFG = ((team_b_fgm + (.5 * team_b_3PT_made))/team_b_fga)

In [317]:
team_b_eFG

0.48214285714285715

Team B's eFG% is 48.21%

In [318]:
team_b_2PT_efg = ((team_b_2PT_made)/team_b_2PT)

In [319]:
team_b_2PT_efg

0.4601226993865031

In [320]:
team_b_C3_efg = ((team_b_C3_made + (.5 * team_b_C3_made))/team_b_C3)

In [321]:
team_b_C3_efg

0.35714285714285715

In [322]:
team_b_NC3_efg = ((team_b_NC3_made + (.5 * team_b_NC3_made))/team_b_NC3)

In [323]:
team_b_NC3_efg

0.546875

On each respective shot, Team A's eFG% was:
    
    2PT - 46.01%
    C3 - 35.71%
    NC3 - 54.69%